util/timers/monotonic_test.go

```diff
@@ -152,3 +153,43 @@ func TestMonotonicEncodeDecode(t *testing.T) {
 		singleTest(c, "random")
 	}
 }
+
+func TestTimeoutTypes(t *testing.T) {
+	partitiontest.PartitionTest(t)
+
+	var m Monotonic
+	var c Clock
+
+	d := time.Millisecond * 100
+
+	c = m.Zero()
+	ch1 := c.TimeoutAt(d, Filter)
+	ch2 := c.TimeoutAt(d, FastRecovery)
+	if polled(ch1) {
```

Comment:
This seems in danger of being a flakey test due to delays/sluggishness on the build server.